internal/api/grpc/oidc/v2/oidc.go

```diff
@@ -73,6 +73,14 @@ func promptToPb(p domain.Prompt) oidc_pb.Prompt {
 	}
 }
 
+func (s *Server) checkPermission(ctx context.Context, clientID string, userID string) error {
+	permission, err := s.query.CheckProjectPermissionByClientID(ctx, clientID, userID)
+	if err != nil || !permission {
+		return zerrors.ThrowPermissionDenied(nil, "OIDC-foSyH49RvL", "Errors.PermissionDenied")
+	}
+	return nil
+}
```

Comment:
we currently provide detailed information (which precheck failed):
https://github.com/zitadel/zitadel/blob/40082745f473945de31bfada638c56372f93e48d/internal/api/ui/login/renderer.go#L332-L335

Generally, i'd name everything more as "authentication checks" rather than "permission".

Also shouldn't it just be a moved to the commands instead of passing it as an argument. We always want to check that. Currently it's already missing on the V2 Beta endpoint.